api/v1alpha1/dnsrecord_types.go

```diff
@@ -53,12 +55,16 @@ type HealthCheckStatusProbe struct {
 }
 
 // DNSRecordSpec defines the desired state of DNSRecord
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.ownerID) || has(self.ownerID)", message="OwnerID can't be unset if it was previously set"
```

Comment:
Why isn't this covered by the rule on line 64? 
`// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="OwnerID is immutable"`

Reply:
This is only checked once it's set see https://kubernetes.io/blog/2022/09/29/enforce-immutability-using-cel/#immutablility-after-first-modification

Also [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#transition-rules) and [here](https://opensource.googleblog.com/2023/11/kubernetes-crd-validation-using-cel.html)

Transition Rule | Purpose
-- | --
self == oldSelf | For a required field, make that field immutable once it is set. For an optional field, only allow transitioning from unset to set, or from set to unset.

Reply:
Lot of messing around trying to get that to work, but if you see a better way... In reality though it doesn't really matter if it gets changed since the controller sets the `ownerID` once and never lets it change again.